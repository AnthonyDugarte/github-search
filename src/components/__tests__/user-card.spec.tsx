import React from "react"
import { render } from "@testing-library/react"

import UserCard from "../user-card"

describe("User card component", () => {
  test("Matches snapshot with search", () => {
    const { asFragment } = render(
      <UserCard
        search="an"
        data={{ username: "Anthony", avatar_url: "", profile_url: "" }}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test("Matches snapshot without search", () => {
    const { asFragment } = render(
      <UserCard
        data={{ username: "Anthony", avatar_url: "", profile_url: "" }}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
