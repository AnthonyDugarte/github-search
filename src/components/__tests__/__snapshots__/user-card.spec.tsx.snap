// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User card component Matches snapshot with search 1`] = `
<DocumentFragment>
  <a
    class="group flex mb-4 last:mb-0 bg-white rounded-l-full rounded-r-full border border-gray-400 hover:outline-none hover:shadow-outline"
    href=""
    target="_blank"
  >
    <img
      class="rounded-full h-24 group-hover:outline-none group-hover:shadow-lg"
      src=""
      title="Anthony"
    />
    <div
      class="flex-1 p-4"
    >
      <div
        class="font-bold text-xl"
      >
        <span
          class="text-red-500"
        >
          An
        </span>
        thony
      </div>
    </div>
  </a>
</DocumentFragment>
`;

exports[`User card component Matches snapshot without search 1`] = `
<DocumentFragment>
  <a
    class="group flex mb-4 last:mb-0 bg-white rounded-l-full rounded-r-full border border-gray-400 hover:outline-none hover:shadow-outline"
    href=""
    target="_blank"
  >
    <img
      class="rounded-full h-24 group-hover:outline-none group-hover:shadow-lg"
      src=""
      title="Anthony"
    />
    <div
      class="flex-1 p-4"
    >
      <div
        class="font-bold text-xl"
      >
        Anthony
      </div>
    </div>
  </a>
</DocumentFragment>
`;
